### PR TITLE
Accept a coro by run_app and gunicorn worker

### PR DESCRIPTION
**Pull request by asvetlov**
#2739

### CODE REVIEW

**Comment by asvetlov** [comment]
@webknjaz please review again

**Comment by asvetlov** [comment]
@webknjaz I would like to get your approval if you agree with the PR

**Comment by webknjaz** [comment]
Sorry, was away from FOSS activities for a while

File: tests/test_run_app.py [hunk 1]
```diff
@@ -516,3 +516,25 @@ def test_startup_cleanup_signals_even_on_failure(patched_loop):
...
+        app = web.Application()
+        startup_handler = make_mocked_coro()
+        app.on_startup.append(startup_handler)
+        cleanup_handler = make_mocked_coro()
+        app.on_cleanup.append(cleanup_handler)
+        return app
+
+    web.run_app(make_app(), print=stopper(patched_loop))
+
+    patched_loop.create_server.assert_called_with(mock.ANY, '0.0.0.0', 8080,
```
Comment (webknjaz) [inline tests/test_run_app.py:535]: @asvetlov are you sure `::1` won't pop-up here in a flaky manner?

Comment (asvetlov) [reply tests/test_run_app.py:535]: Sorry, don't follow.
Please elaborate

**Pull request merged**